src/tensors/abstracttensor.jl

```diff
@@ -196,23 +186,36 @@ function adjointtensorindices(t::AbstractTensorMap, p::Index2Tuple)
     return (adjointtensorindices(t, p[1]), adjointtensorindices(t, p[2]))
 end
 
-@doc """
-    blocks(t::AbstractTensorMap) -> SectorDict{<:Sector,<:DenseMatrix}
+# tensor characteristics: work on instances and pass to type
+#------------------------------------------------------------
+spacetype(t::AbstractTensorMap) = spacetype(typeof(t))
+sectortype(t::AbstractTensorMap) = sectortype(typeof(t))
+InnerProductStyle(t::AbstractTensorMap) = InnerProductStyle(typeof(t))
+field(t::AbstractTensorMap) = field(typeof(t))
+storagetype(t::AbstractTensorMap) = storagetype(typeof(t))
+similarstoragetype(t::AbstractTensorMap, TT) = similarstoragetype(typeof(t), TT)
 
-Return an iterator over all blocks of a tensor, i.e. all coupled sectors and their
-corresponding blocks.
+numout(t::AbstractTensorMap) = numout(typeof(t))
+numin(t::AbstractTensorMap) = numin(typeof(t))
+numind(t::AbstractTensorMap) = numind(typeof(t))
 
-See also [`block`](@ref), [`blocksectors`](@ref), [`blockdim`](@ref) and [`hasblock`](@ref).
+# tensor characteristics: data structure and properties
+#------------------------------------------------------
 """
-blocks(t::AbstractTensorMap) = SectorDict(c => block(t, c) for c in blocksectors(t)) # TODO: make iterator
+    fusionblockstructure(t::AbstractTensorMap) -> TensorStructure
 
-@doc """
-    block(t::AbstractTensorMap, c::Sector) -> DenseMatrix
+Return the necessary structure information to decompose a tensor in blocks labeled by
+coupled sectors and in subblocks labeled by a splitting-fusion tree couple.
+"""
+fusionblockstructure(t::AbstractTensorMap) = fusionblockstructure(space(t))
```

Comment:
Am I correct in assuming that this really is tied to `TensorMap`, and not necessarily to `AbstractTensorMap`? Would this still be correct for `AdjointTensorMap`?

Reply:
It depends on what you mean? The functions still works as it just acts on the underlying `HomSpace` instance, but of course the information it retrieves is only really useful `TensorMap` instances.